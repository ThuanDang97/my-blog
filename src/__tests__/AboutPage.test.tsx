// Components
import AboutPage from '@pages/about'

// Utils
import { render } from '@utils/testUtils'

// Mocks Data
import { MOCK_DATA_ABOUT_PAGE } from '@constants/__mock__/mockData'

describe('Page About', () => {
  test('Page About matches DOM Snapshot', () => {
    const { container } = render(<AboutPage recipes={MOCK_DATA_ABOUT_PAGE} />)

    expect(container).toMatchSnapshot()
  })
})
