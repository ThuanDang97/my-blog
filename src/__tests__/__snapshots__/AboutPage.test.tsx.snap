// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page About Page About matches DOM Snapshot 1`] = `
<div>
  <header
    class="MuiBox-root css-1p2cbqg"
  >
    <div
      class="MuiBox-root css-kp6j5k"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-9tjvhq-MuiContainer-root"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-1pidjnp-MuiTypography-root"
        >
          <a
            href="/"
          >
            BEAUTYONCODE
          </a>
        </h1>
        <div
          class="MuiBox-root css-7w1jqr"
        >
          <ul
            class="MuiList-root MuiList-padding css-127t676-MuiList-root"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1rk1jkp-MuiListItem-root"
            >
              <a
                href="/"
              >
                HOME
              </a>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1rk1jkp-MuiListItem-root"
            >
              <a
                href="/category/most-views"
              >
                Most views
              </a>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1rk1jkp-MuiListItem-root"
            >
              <a
                href="/blog"
              >
                Blog
              </a>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1rk1jkp-MuiListItem-root"
            >
              <a
                href="/about"
              >
                About
              </a>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1rk1jkp-MuiListItem-root"
            >
              <a
                href="/category/short-posts"
              >
                Short Posts
              </a>
            </li>
          </ul>
          <button
            aria-label="menu"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeMedium css-bxr13g-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="MenuIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-label="search"
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-15on3wz-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1unfims-MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="SearchIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div>
          loading.....
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-fcpxmd"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
      >
        <nav
          class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
        >
          <ol
            class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
          >
            <li
              class="MuiBreadcrumbs-li"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-7xpsn0-MuiTypography-root"
              >
                <a
                  href="/"
                >
                  HOME
                </a>
              </p>
            </li>
            <li
              aria-hidden="true"
              class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
            >
              /
            </li>
            <li
              class="MuiBreadcrumbs-li"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-hiwasb-MuiTypography-root"
              >
                <a
                  href="/about"
                >
                  about
                </a>
              </p>
            </li>
          </ol>
        </nav>
        <h2
          class="MuiTypography-root MuiTypography-h2 css-19qx9ri-MuiTypography-root"
          data-testid="page-title"
        >
          About
        </h2>
      </div>
    </div>
  </header>
  <main
    class="MuiBox-root css-0"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
    >
      <main
        class="MuiBox-root css-0"
      >
        <section
          class="MuiBox-root css-0"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 css-ymhh61-MuiTypography-root"
          >
            Cám ơn bạn đã ghé thăm blog BeautyOnCode! Hãy để tôi làm quen với bạn bằng câu chuyện của tôi và của blog này bằng lời kể của mình nha.
          </h5>
          <h2
            class="MuiTypography-root MuiTypography-h2 css-9nz9xo-MuiTypography-root"
          >
            1. TO 
            Thanh Nguyễn
          </h2>
          <img
            alt="avatar"
            data-nimg="1"
            decoding="async"
            height="200"
            loading="eager"
            src="/_next/image?url=https%3A%2F%2Fimages.ctfassets.net%2Font2c8ajcvde%2F2q42Qytnxy63qAgEQgSG67%2F7ad54dfe423bf8253b7e2094ece12262%2Favatar.png&w=640&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fimages.ctfassets.net%2Font2c8ajcvde%2F2q42Qytnxy63qAgEQgSG67%2F7ad54dfe423bf8253b7e2094ece12262%2Favatar.png&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fimages.ctfassets.net%2Font2c8ajcvde%2F2q42Qytnxy63qAgEQgSG67%2F7ad54dfe423bf8253b7e2094ece12262%2Favatar.png&w=640&q=75 2x"
            style="color: transparent;"
            width="200"
          />
          <div
            class="MuiBox-root css-gmwf1n"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-19cn5b7-MuiTypography-root"
            >
              Xin chào, tôi là 
              <b>
                <i>
                  Thanh Nguyễn (Nguyễn Diễm Thanh)
                </i>
              </b>
               – tác giả của blog 
              <i>
                BeautyOnCode
              </i>
              .Tôi sinh ra và lớn lên ở Đà Nẵng và hiện cũng đang sống và làm việc tại đây. Tôi từng nghĩ mình sẽ trở thành nhà thiết kế đồ họa nhưng nghề lập trình đã chọn mình. Hiện tôi đang làm 
              <b>
                Web Developer tại 
              </b>
              <a
                href="https://www.agilityio.com/"
              >
                <u>
                  <i>
                    <b>
                      AgilityIO
                    </b>
                  </i>
                </u>
              </a>
               một công ty có văn phòng ở NewYork, Mỹ và đội ngũ phát triển ở Việt Nam. Nếu bạn muốn tìm hiểu thêm về công ty này, có thể xem thêm các video: 
              <a
                href="https://www.youtube.com/watch?v=EwOuGsrYmGA"
              >
                <i>
                  <u>
                    AgilityIO
                  </u>
                </i>
              </a>
              <i>
                , 
              </i>
              <a
                href="https://www.youtube.com/watch?v=ci9QZbNLct0"
              >
                <i>
                  <u>
                    Life At AgilityIO
                  </u>
                </i>
              </a>
              <i>
                . 
              </i>
              Công việc của tôi xoay quanh việc lập trình, sử dụng công nghệ như một giải pháp cùng khách hàng của mình tạo ra các sản phẩm tốt cho người dùng.  
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-19cn5b7-MuiTypography-root"
            >
              Trong quá trình học lập trình từ con số 0 của mình, tôi nhận ra nhiều điều thú vị, tôi đã nhìn lập trình ở góc nhìn của một người thiết kế sản phẩm(designer), của một người phân tích sản phẩm(business analyst) và của người dùng(user). Tôi thấy bản thân họ là mình và mình cũng là họ, chúng tôi có những mối tương quan sâu sắc và có cùng một mục tiêu, là sử dụng công nghệ để giải quyết các vấn đề. Lập trình rõ là không dễ, và tôi thích đều đó, tôi thích tính chất của công việc này, đó là sự thay đổi không ngừng, sự học không ngừng nghỉ và nó dạy cho tôi thật nhiều về sự khiêm tốn.
Trong cuộc sống, tôi cũng đang có một hành trình vô cùng thú vị, từ một con bé tomboy cá tính trở thành một người phụ nữ lập gia đình và có con. Tôi cũng đã nuôi một ước mơ từ thời sinh viên là vừa có thể kiếm tiền trang trải cuộc sống vừa có thể được làm thiện nguyện cống hiến cho cộng đồng, có thể là làm cho một tổ chức phi chính phủ với công việc gì đó chẳng hạn. Tuy đã lâu chưa từng nhắc đến, nhưng nó vẫn cứ ở trong mình, đây cũng là lý do tôi chọn làm 
              <b>
                tình nguyện viên cho
              </b>
               
              <a
                href="https://steamforvietnam.org/"
              >
                <u>
                  <i>
                    <b>
                      STEAM for Viet Nam
                    </b>
                  </i>
                </u>
              </a>
               khi đủ duyên đến.Về viết lách, tôi bắt đầu viết từ rất lâu rồi, từ cấp ba tôi đã có cuốn truyện riêng của mình tự viết, nhưng đã bị thất lạc. Tôi thỉnh thoảng rất hay thích viết cho mình, như là một cách để rà soát và cảm nhận, nhưng chưa từng nghiêm túc viết. Sau khi đọc cuốn 
              <a
                href="https://beautyoncode.com/sach-khi-hoi-tho-hoa-thinh-khong-paul-kalanithi/"
              >
                <i>
                  <u>
                    “Khi hơi thở hoá thinh không”
                  </u>
                </i>
              </a>
              , tôi thấy cuộc sống quả là vô thường và bắt đầu suy nghĩ nghiêm túc hơn cho một nơi mà tôi có thể được sống với chính mình, tự học mà không bị ai bắt ép và có thể thoả mái chia sẻ với mình và với mọi người. Đây cũng là các đề tài chủ yếu của blog 
              <i>
                BeautyOnCode
              </i>
              .Sau khi tạm chia tay STEAM for Viet Nam, tôi bắt đầu viết tự do nhiều hơn và hiện đang cộng tác cùng 
              <a
                href="https://careerly.vn/profiles/1140"
              >
                <b>
                  Careerly
                </b>
              </a>
              , là ứng dụng kết nối mạng lưới người Việt làm tại các công ty công nghệ, nơi mình chia sẻ đến bạn đọc các bài viết ngắn về tech.
Bạn có thể liên hệ với tôi qua email beautyoncode@gmail.com hoặc 
              <a
                href="https://www.facebook.com/beautyoncode/"
              >
                <u>
                  <i>
                    fanpage BeautyOnCode
                  </i>
                </u>
              </a>
              .
            </p>
          </div>
          <h2
            class="MuiTypography-root MuiTypography-h2 css-1ybkhuw-MuiTypography-root"
          >
            2. TO 
            beautyoncode
          </h2>
          <div
            class="MuiBox-root css-gmwf1n"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-19cn5b7-MuiTypography-root"
            >
              <b>
                BeautyOnCode
              </b>
               là blog viết về lập trình và về những câu chuyện trong cuộc sống của tôi (Thanh) với các đề tài như 
              <a
                href="https://beautyoncode.com/category/lap-trinh/"
              >
                <i>
                  <u>
                    lập trình
                  </u>
                </i>
              </a>
              , 
              <a
                href="https://beautyoncode.com/category/hanh-trinh-cua-toi/yeu-ban-than/"
              >
                <u>
                  <i>
                    làm blog
                  </i>
                </u>
              </a>
              , 
              <a
                href="https://beautyoncode.com/category/hanh-trinh-cua-toi/yeu-ban-than/"
              >
                <u>
                  <i>
                    hành trình phát triển bản thân
                  </i>
                </u>
              </a>
              , … Hiện BeautyOnCode đang có gần 90 bài viết, bấm vào 
              <a
                href="https://beautyoncode.com/category/most-views/"
              >
                “Xem Nhiều Nhất”
              </a>
               để đọc các bài viết được đọc giả đọc nhiều nhất. Bạn cũng có thể ghé 
              <a
                href="https://beautyoncode.com/muc-luc/"
              >
                “Mục lục”
              </a>
               để xem các bài viết được chia theo các chuỗi bài cho từng loại chủ đề (Python, Linux, JavaScript, Leetcode, …). Bên cạnh đó, hơn 62 bài viết ở mục 
              <a
                href="https://beautyoncode.com/category/short-posts/"
              >
                “Bài Viết Ngắn”
              </a>
              , là những góc nhìn về công nghệ, giới thiệu các tin tức, nguồn tài liệu, … Và cuối cùng là một 
              <a
                href="https://beautyoncode.com/podcast/"
              >
                podcast
              </a>
               với hơn 60 episodes về các bài viết tự do, các bài viết từ blog khác và sách.Về 
              <i>
                <b>
                  tên blog này
                </b>
              </i>
              , nó có nhiều ý nghĩa khác nhau. Trước hết, “
              <b>
                BeautyOnCode
              </b>
              ” là bút danh tôi tự đặt cho mình, cho nó ngầu ấy mà, ý là một người phụ nữ trong ngành lập trình, tôi muốn viết về trải nghiệm của mình trong lĩnh vực này. Ngoài ra, “
              <b>
                BeautyOnCode
              </b>
              ” còn có ý nghĩa là nét đẹp của code nữa, tôi thấy nó đẹp vì nghề này, không mang nghiệp nặng gì, chỉ đơn thuần là làm việc trên máy tính, tôi còn thấy đa phần mọi người làm trong ngành đến hướng thiện, đều là những người tốt nên tôi thấy danh xưng này rất là xứng đáng.
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-19cn5b7-MuiTypography-root"
            >
              Nếu bạn yêu thích BeautyOnCode, bạn có thể đăng ký theo dõi blog ở form bên dưới hoặc trang ở 
              <a
                href="http://eepurl.com/hJr_g9"
              >
                <i>
                  <u>
                    đây
                  </u>
                </i>
              </a>
              , hoặc follow 
              <a
                href="https://www.facebook.com/beautyoncode/"
              >
                <i>
                  <u>
                    fanpage
                  </u>
                </i>
              </a>
               này nha.Bạn cũng có thể cùng mình duy trì blog này bằng cách ủng hộ mình ở trang 
              <a
                href="https://beautyoncode.com/support/"
              >
                Support Me
              </a>
              .
              <i>
                Chúc bạn một ngày tốt lành!
              </i>
              Thanh Nguyễn
            </p>
          </div>
        </section>
      </main>
    </div>
  </main>
</div>
`;
